docs/crawl_data_parsing.md

```diff
@@ -0,0 +1,97 @@
+
+## What crawled sites are included as metadata records?
+
+A browsertrix web crawl generates a bunch of files as the result of the crawl:
+
+```text
+├── archive
+│   ├── rec-20230929152646786455-3e0077b1f009.warc.gz
+│   ├── rec-20230929152647253458-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648227041-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648249931-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648436859-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648567703-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648603443-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648639794-3e0077b1f009.warc.gz
+│   ├── rec-20230929152649186320-3e0077b1f009.warc.gz
+│   └── rec-20230929152649791199-3e0077b1f009.warc.gz
+├── homepage.wacz
+├── indexes
+│   └── index.cdxj
+├── logs
+│   └── crawl-20230929152634980.log
+├── pages
+│   └── pages.jsonl
+├── static
+└── templates
+```
+
+Note the [WACZ](https://replayweb.page/docs/wacz-format) file here, `homepage.wacz` (where "homepage" is an arbitrary `--crawl-name`).  This file is a compressed form of the entire crawl, and is the only asset parsed by this application when generating metadata records.  This is handy for multiple reasons:
+  * this file can be saved or copied, representing the crawl in its totality
+  * parsing data from the crawl is only concerned with a single file as the entrypoint (though multiple files are later accessed inside) allowing creation of metadata records even from a remote file in S3
+
+The structure of the WACZ file is nearly identical to the uncompressed crawl results, with some minor, but important, additions:
+```text
+.
+├── archive
+│   ├── rec-20230929152646786455-3e0077b1f009.warc.gz
+│   ├── rec-20230929152647253458-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648227041-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648249931-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648436859-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648567703-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648603443-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648639794-3e0077b1f009.warc.gz
+│   ├── rec-20230929152649186320-3e0077b1f009.warc.gz
+│   └── rec-20230929152649791199-3e0077b1f009.warc.gz
+├── datapackage-digest.json
+├── datapackage.json
+├── indexes
+│   ├── index.cdx.gz
+│   └── index.idx
+├── logs
+│   └── crawl-20230929152634980.log
+└── pages
+    ├── extraPages.jsonl
+    └── pages.jsonl
+```
+
+The WACZ files `pages/pages.jsonl` and `pages/extraPages.jsonl` are what create the canonical list of websites to be considered for metadata records.  While a web crawl may make hundreds of HTTP requests to gather Javascript, CSS, JSON, or any other supporting files, what we are most interested in are HTML pages (websites) that were directly included as seeds or discovered via the crawl configurations.
```

Comment:
When we run `make test-harvest-local`, is `homepage.wacz` decompressed in the `output/` folder?

Reply:
Technically, not quite.  These [two methods](https://github.com/MITLibraries/browsertrix-harvester/blob/pr3-parser/harvester/wacz.py#L81-L105) in the `WACZClient` are important, as they support reading files from the WACZ zip archive without ever unzipping it.  

The WACZ file will be available in the host machine `./output/crawls/collections/<CRAWL_NAME>/<CRAWL_NAME>.wacz` folder, as the container path `/crawls` is mounted to host `./output/`.

And, including the `--wacz-output-file` when calling the `harvest` command, can optionally _also_ write that file somewhere else.

In summary:
  * the WACZ file is never unzipped by any normal operations of this app
  * methods exist to read these files from the zipped archive
  * the WACZ is mounted to `./output` for local dev work, but it's the `--wacz-output-file` CLI argument that would be helpful for saving that file somewhere in a normal run